hera_cal/reflections.py

```diff
@@ -0,0 +1,516 @@
+                       broadcast_flags=False, time_thresh=0.05, overwrite=False, verbose=False):
+        """
+        Run a Delay Clean on self.data dictionary to derive a model of the
+        visibility free of flagged channels. CLEAN data is inserted into
+        self.clean_data.
+
+        Args:
+            keys : len-3 tuple, keys of self.data to run clean on. Default is all keys.
+            tol : float, stopping tolerance for CLEAN. See aipy.deconv.clean
+            maxiter : int, maximum number of CLEAN iterations
+            gain : float, CLEAN gain
+            skip_wgt : float, fraction of flagged channels needed to skip a time integration
+            dly_cut : float, maximum delay [nanoseconds] to model FT of CLEAN visibility
+            edgecut : int, number of channels to exclude in CLEAN on either side of band.
+                Note this is not the same as flagging edge channels. This excludes them
+                entirely, such that if a taper is applied, it is not discontinuous at
+                the band edges.
+            taper : str, Tapering function to apply across freq before FFT
+            alpha : float, if taper is Tukey, this its alpha parameter
+            timeavg : bool, if True, average data across time weighted by flags
+            broadcast_flags : bool, if True, broadcast flags across time using time_thresh
+            time_thresh : float, if fraction of flagged times exceeds this ratio,
+                flag a channel for all times.
+        """
+        # initialize containers
+        if hasattr(self, "clean_data") and not overwrite:
+            raise ValueError("self.clean_data exists and overwrite is False...")
+        self.clean_data = DataContainer({})
+        self.resid_data = DataContainer({})
+        self.clean_info = {}
+        self.clean_freqs = self.freqs
+        if edgecut > 0:
+            self.clean_freqs = self.clean_freqs[edgecut:-edgecut]
+
+        # get keys
+        if keys is None:
+            keys = self.data.keys()
+
+        # iterate over keys
+        for k in keys:
+            echo("...Cleaning data key {}".format(k), verbose=verbose)
+            (model, residual, dlys,
+             info) = delay_filter(self.data[k].copy(), self.flags[k].copy(), self.dnu, tol=tol,
+                                  maxiter=maxiter, gain=gain, skip_wgt=skip_wgt, dly_cut=dly_cut,
+                                  edgecut=edgecut, taper=taper, alpha=alpha, timeavg=timeavg,
+                                  broadcast_flags=broadcast_flags, time_thresh=time_thresh)
+            self.clean_data[k] = model
+            self.resid_data[k] = residual
+            self.clean_info[k] = info
+
+        self.clean_dlys = dlys
+        self.clean_times = self.times
+        if timeavg:
+            self.clean_times = np.mean(self.clean_times, keepdims=True)
+
+    def model_reflections(self, dly_range, keys=None, data='clean', edgecut=0, taper='none',
```

Comment:
Hey, yeah the phase fitting can also be done in delay space, but I was having some issues with the implementation. Anyways, because one has run a delay clean, one can manually flag problematic channels (I assume the MWA band gaps were flagged data...) and it should handle it just fine. The other thing is that by specifying a delay range, you can isolate a cable reflection of approximately known delay from other things in the data..